### PR TITLE
feat: add proxy plugin to transfer messages from one backend to multiple backends

### PR DESCRIPTION
**Pull request by yglsaltfish**

### CODE REVIEW

File: src/examples/plugins/CMakeLists.txt [hunk 1]
```diff
@@ -61,3 +61,11 @@ endif()
...
   add_subdirectory(grpc_plugin)
 endif()
+
+if(AIMRT_BUILD_PROXY_PLUGIN
+   AND AIMRT_BUILD_WITH_PROTOBUF
+   AND AIMRT_BUILD_WITH_ROS2
+   AND AIMRT_BUILD_WITH_ROS2_PLUGIN
+   AND AIMRT_BUILD_ZENOH_PLUGIN)
+  add_subdirectory(proxy_plugin)
+endif()
```
Comment (zhangyi1357) [inline src/examples/plugins/CMakeLists.txt:71]: Add Those dependency options into root CMakeLists.txt's cmake_dependent_options

File: src/plugins/proxy_plugin/topic_meta_key.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+             (std::hash<std::string>()(k.msg_type));
+    }
+  };
+};
+
+struct TopicMeta {
+  std::string topic_name;
+  std::string msg_type;
+  std::string serialization_type;
+  std::vector<std::string> pub_topic_name;
```
Comment (zhangyi1357) [inline src/plugins/proxy_plugin/topic_meta_key.h:31]: Seems there are many TopicMeta struct definitions in the repo. Can we abstract their commonalities for some degree of reuse?

File: src/plugins/proxy_plugin/proxy_plugin.cc [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+  }
+  type_support_pkg_loader_vec_.emplace_back(std::move(loader_ptr));
+  AIMRT_TRACE("Load {} type support pkgs.", type_support_pkg_loader_vec_.size());
+}
+
+void ProxyPlugin::RegisterSubChannel() {
+  using namespace aimrt::runtime::core::channel;
+
+  for (auto& proxy_action_itr : proxy_action_map_) {
+    auto& proxy_action = *(proxy_action_itr.second);
```
Comment (zhangyi1357) [inline src/plugins/proxy_plugin/proxy_plugin.cc:201]: Structured bindings can eliminate these poorly readable `second`.

File: src/examples/plugins/CMakeLists.txt (hunk 1 above)
Comment (yglsaltfish) [reply src/examples/plugins/CMakeLists.txt:71]: here is example , just need to add_subdirectory

File: src/plugins/proxy_plugin/proxy_plugin.cc (hunk 3 above)
Comment (yglsaltfish) [reply src/plugins/proxy_plugin/proxy_plugin.cc:196]: I think the name itself is readble enough , here is just an pointer to obj ProxyAction

Comment (zhangyi1357) [reply src/plugins/proxy_plugin/proxy_plugin.cc:196]:
```cpp
for (auto& proxy_action_itr : proxy_action_map_) {
    auto& proxy_action = *(proxy_action_itr.second);
```

```cpp
for (auto& [_, proxy_action_ptr] : proxy_action_map_) {
    auto& proxy_action = *proxy_action_ptr;
```

The following approach allows readers to understand the type stored inside proxy_action_map_ without inspecting its contents.

File: src/plugins/proxy_plugin/topic_meta_key.h (hunk 2 above)
Comment (yglsaltfish) [reply src/plugins/proxy_plugin/topic_meta_key.h]: i thing we can abstract topic_meta_key

File: src/plugins/proxy_plugin/proxy_plugin.cc (hunk 3 above)
Comment (yglsaltfish) [reply src/plugins/proxy_plugin/proxy_plugin.cc]: Amazing! It's true that it's simpler

File: document/sphinx-cn/tutorials/plugins/proxy_plugin.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          options:
+            node_name: example_ros2_pb_chn_publisher_node
+            executor_type: MultiThreaded # SingleThreaded/StaticSingleThreaded/MultiThreaded
+            executor_thread_num: 2
+  log:
+    core_lvl: Info # Trace/Debug/Info
+    backends:
+      - type: console
+  channel:
+    # ...
```
Comment (AimRT) [inline document/sphinx-cn/tutorials/plugins/proxy_plugin.md:63]: add channel cfg for topic

**Pull request merged**